### ISSUE TITLE
Mistakes in the code

### ISSUE THREAD

**Issue by BBerabi** [lin-tan/CoCoNut-Artifact#3]
Hi, 

I have been looking at your code for the last few days. The code you pushed as an artifact clearly does not implement the architecture you proposed in the paper.

In the line **421** of file **fairseq-context/fairseq/models/fconv.py** you define the forward function of your FConvContextEncoder with arguments `src_tokens, src_length, ctx_tokens, ctx_length`, but you never use the the parameters related to context! In fact, you define a context encoder in your __init__() method but in the forward method, you have the following lines.

```
src_output = self.input_encoder.forward(src_tokens,src_lengths)
ctx_output = self.input_encoder.forward(src_tokens,src_lengths)

```
So, your input and context encoder both take source tokens and there is no difference between them!!
If one gives the correct inputs to each encoder, then the code later crashes as the different sequences lengths are not handled in your torch.cat statement. They have to padded before concatenation. Also the encoder mask has to be chosen in a proper way...

**Comment by thibolu** [comment]
Thank you for finding this issue. 
After reviewing the code, we agree this is a copy-paste bug.
We are actively working on an update, fixing the code, and retraining the context-aware models to make sure everything works correctly. 

Thanks again @BBerabi, we will keep you updated.

**Comment by thibolu** [comment]
The mistake has been fixed.
